src/components/metadataSelects/CategoryComboSelect.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useCallback, useRef, useState } from 'react'
+import { SelectOption } from '../../types'
+import { SearchableSingleSelect } from '../SearchableSingleSelect'
+import { useInitialOptionQuery } from './useInitialOptionQuery'
+import { useOptionsQuery } from './useOptionsQuery'
+
+function computeDisplayOptions({
+    selected,
+    selectedOption,
+    options,
+}: {
+    options: SelectOption[]
+    selected?: string
+    selectedOption?: SelectOption
+}): SelectOption[] {
+    // This happens only when we haven't fetched the lable for an initially
+    // selected value. Don't show anything to prevent error that an option is
+    // missing
+    if (!selectedOption && selected) {
+        return []
+    }
+
+    const optionsContainSelected = options?.find(
+        ({ value }) => value === selected
+    )
+
+    if (selectedOption && !optionsContainSelected) {
+        return [...(options as SelectOption[]), selectedOption as SelectOption]
```

Comment:
This type-cast is not needed for me? I get no errors with ` return [...options, selectedOption]`